compiler/quilt/tools/command.py

```diff
@@ -259,14 +260,20 @@ def _match_hash(session, owner, pkg, hash, raise_exception=True):
             pkg=pkg
         )
     )
-    for entry in reversed(response.json()['logs']):
-        # support short hashes
-        if entry['hash'].startswith(hash):
-            return entry['hash']
 
+    matches = set(entry['hash'] for entry in response.json()['logs']
+                  if entry['hash'].startswith(hash))
+
+    if len(matches) == 1:
+        return matches.pop()
+    if len(matches) > 1:
+        # Sorting for consistency in testing, as well as visual comparison of hashes
+        ambiguous = '\n'.join(sorted(matches))
+        raise CommandException(
```

Comment:
It seems like this should also check for `raise_exception`... But it looks like nobody uses that param, so perhaps we should delete it. @asah - you added it, right?

Reply:
fine to remove it.

Reply:
I thought about that, but then I wasn't sure if it was sensical to return None when multiple hashes were indeed identified, which is pretty much a show-stopper.  I figured anyone using it would notice an exception, but not notice the problem if it was squelched into just returning None.

..removed the raise_exception param.  None is no longer returned, and an exception is raised in all cases where the number of matches isn't 1.